components/suggest/src/rs.rs

```diff
@@ -339,3 +369,48 @@ where
 {
     String::deserialize(deserializer).map(|s| s.parse().ok())
 }
+
+impl SuggestAttachment for DownloadedAmpWikipediaSuggestion {
+    fn unknown_fields(&self) -> &serde_json::Map<String, serde_json::Value> {
+        match self {
+            Self::Amp(DownloadedAmpSuggestion { unknown_fields, .. }) => unknown_fields,
+            Self::Wikipedia(DownloadedWikipediaSuggestion { unknown_fields, .. }) => unknown_fields,
+        }
+    }
+}
+
+impl SuggestAttachment for DownloadedAmoSuggestion {
```

Comment:
Looks like most `Downloaded()Suggestionn` types share the same implementation, what do you think about providing a macro for that? Please disregard if macros are not a thing for you.

```rs
macro_rules! impl_suggest_attachment {
    ( 
      $(
        $t:ident
      ),+
    ) => {
        $(
          impl SuggestAttachment for $t {
             fn unknown_fields(&self) -> &serde_json::Map<String, serde_json::Value> {
                 &self.unknown_fields
             }
          }
        )+
    }
}

impl_suggest_attachment!(
  DownloadedAmoSuggestion,
  DownloadedPocketSuggestion,
  ...
);
```